From b5c829f32fbc4f3e334e5aeed9d8fa674c0b52b5 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@freescale.com>
Date: Mon, 25 May 2015 10:57:36 +0800
Subject: [PATCH 1127/1594] MLK-10964 input: imx6ul_tsc: ignore the last touch
 event

commit bab444582b3fe18add229b74fa7cc6728a40d126 from
git://git.freescale.com/imx/linux-2.6-imx.git

When the touch screen be touched, it will raise the interrupt
for many times. In the last interrupt, we should ignor the X, Y
value, and don't report them, because it is inaccurate.

Signed-off-by: Haibo Chen <haibo.chen@freescale.com>
---
 drivers/input/touchscreen/imx6ul_tsc.c |    9 ++++-----
 1 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/input/touchscreen/imx6ul_tsc.c b/drivers/input/touchscreen/imx6ul_tsc.c
index 11a8c58..5a5a368 100644
--- a/drivers/input/touchscreen/imx6ul_tsc.c
+++ b/drivers/input/touchscreen/imx6ul_tsc.c
@@ -226,9 +226,6 @@ static irqreturn_t tsc_irq(int irq, void *dev_id)
 		x = (tsc->value >> 16) & 0x0fff;
 		y = tsc->value & 0x0fff;
 
-		input_report_abs(tsc->input, ABS_X, x);
-		input_report_abs(tsc->input, ABS_Y, y);
-
 		/*
 		 * Delay some time(max 2ms), wait the pre-charge done.
 		 * After the pre-change mode, TSC go into detect mode.
@@ -251,9 +248,11 @@ static irqreturn_t tsc_irq(int irq, void *dev_id)
 
 		xnur = gpio_get_value(tsc->xnur_gpio);
 touch_event:
-		if (xnur == 0)
+		if (xnur == 0) {
 			input_report_key(tsc->input, BTN_TOUCH, 1);
-		else
+			input_report_abs(tsc->input, ABS_X, x);
+			input_report_abs(tsc->input, ABS_Y, y);
+		} else
 			input_report_key(tsc->input, BTN_TOUCH, 0);
 
 		input_sync(tsc->input);
-- 
1.7.5.4

